### PR TITLE
tests: use smoke intergration tests for macOS

### PR DESCRIPTION
**Pull request by syu-w**

### CODE REVIEW

**Review by mr-cal — APPROVED** [review]
Nice work!

File: tests/integration/multipass/test_basic.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+        assert proc.stdout == test_file.read_bytes()
+
+        instance.unmount(target=target)
+        assert instance.is_mounted(host_source=home_tmp_path, target=target) is False
+
+        # test stop instance
+        instance.stop()
+        assert instance.is_running() is False
+    finally:
+        instance.delete()
```
Comment (tigarmo) [inline tests/integration/multipass/test_basic.py:105]: minor but there's an alternative to a huge try/finally block like this in pytest tests: add the `request` fixture and then do `request.addfinalizer(instance.delete)` after creating the instance

File: tox.ini [hunk 2]
```diff
@@ -53,7 +53,19 @@ labels =
 change_dir =
     unit: tests/unit
     integration: tests/integration
-commands = pytest {tty:--color=yes} --cov --cov-report=xml:{tox_root}/results/coverage-{env_name}.xml --junit-xml={tox_root}/results/test-results-{env_name}.xml {posargs}
+commands =
+    unit: pytest {tty:--color=yes} --cov --cov-report=xml:{tox_root}/results/coverage-{env_name}.xml --junit-xml={tox_root}/results/test-results-{env_name}.xml {posargs}
+    integration: pytest {tty:--color=yes} --cov --cov-report=xml:{tox_root}/results/coverage-{env_name}.xml --junit-xml={tox_root}/results/test-results-{env_name}.xml -m "not smoketest" {posargs}
```
Comment (tigarmo) [inline tox.ini:58]: how come we don't want to run the smoketest in non-macos?

**Review by tigarmo — APPROVED** [review]
thanks! Left some minor notes but feel free to ignore

File: tests/integration/multipass/test_basic.py (hunk 1 above)
Comment (syu-w) [reply tests/integration/multipass/test_basic.py:105]: I think I cannot call `assert instance.exists() is False` if use that?

File: tox.ini (hunk 2 above)
Comment (syu-w) [reply tox.ini:58]: Because that smoketest only for macos? The integration tests in linux usually works fine.

**Pull request merged**